### ISSUE TITLE
Could not resolve artifact jakarta.xml.ws:jakarta.xml.ws-api::jar

### ISSUE THREAD

**Issue by DMaxter** [quarkiverse/quarkus-cxf#916]
Hello!

I'm currently using Quarkus 2.14.0.Final (not yet upgraded to Quarkus 3) with gradle.
I need to interact with a SOAP API so I just added `implementation("io.quarkiverse.cxf:quarkus-cxf")` to the dependencies of my project. 
I followed the steps in [Your first SOAP Client on Quarkus](https://quarkiverse.github.io/quarkiverse-docs/quarkus-cxf/dev/user-guide/first-soap-client.html) and [Generate the Model classes from WSDL](https://quarkiverse.github.io/quarkiverse-docs/quarkus-cxf/dev/user-guide/generate-java-from-wsdl.html), but when trying to run Quarkus (without adding any code, just trying to obtain the generated classes) I got this exception.

```
Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.index.ApplicationArchiveBuildStep#build threw an exception: java.lang.RuntimeException: Could not resolve artifact jakarta.xml.ws:jakarta.xml.ws-api::jar among the runtime dependencies of the application
        at io.quarkus.deployment.index.ApplicationArchiveBuildStep.addIndexDependencyPaths(ApplicationArchiveBuildStep.java:188)
        at io.quarkus.deployment.index.ApplicationArchiveBuildStep.scanForOtherIndexes(ApplicationArchiveBuildStep.java:151)
        at io.quarkus.deployment.index.ApplicationArchiveBuildStep.build(ApplicationArchiveBuildStep.java:108)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)

        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:335)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:252)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:60)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:86)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:447)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:59)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:149)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:104)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.index.ApplicationArchiveBuildStep#build threw an exception: java.lang.RuntimeException: Could not resolve artifact jakarta.xml.ws:jakarta.xml.ws-api::jar among the runtime dependencies of the application
        at io.quarkus.deployment.index.ApplicationArchiveBuildStep.addIndexDependencyPaths(ApplicationArchiveBuildStep.java:188)
        at io.quarkus.deployment.index.ApplicationArchiveBuildStep.scanForOtherIndexes(ApplicationArchiveBuildStep.java:151)
        at io.quarkus.deployment.index.ApplicationArchiveBuildStep.build(ApplicationArchiveBuildStep.java:108)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)

        at io.quarkus.builder.Execution.run(Execution.java:123)
        at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:331)
        ... 9 more
Caused by: java.lang.RuntimeException: Could not resolve artifact jakarta.xml.ws:jakarta.xml.ws-api::jar among the runtime dependencies of the application
        at io.quarkus.deployment.index.ApplicationArchiveBuildStep.addIndexDependencyPaths(ApplicationArchiveBuildStep.java:188)
        at io.quarkus.deployment.index.ApplicationArchiveBuildStep.scanForOtherIndexes(ApplicationArchiveBuildStep.java:151)
        at io.quarkus.deployment.index.ApplicationArchiveBuildStep.build(ApplicationArchiveBuildStep.java:108)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

Any ideas on how to debug this further or even fix it?

**Comment by shumonsharif** [comment]
The 'dev' version of the docs are currently Quarkus 3 specific.  You may want to try switch to the 1.5 version of the docs as described [here](https://github.com/quarkiverse/quarkus-cxf#documentation) and take a look specifically at the [dependency management](https://quarkiverse.github.io/quarkiverse-docs/quarkus-cxf/1.5/user-guide/create-project.html#dependency-management) section.

Please feel free to attach a small reproducer if you may need further help!

**Comment by DMaxter** [comment]
That section doesn't seem to affect my project as I'm using gradle and it is for Maven.

Anyway, I have created a super simple reproducer (just added the quarkus-cxf dependency to an empty project)
[reproducer.zip](https://github.com/quarkiverse/quarkus-cxf/files/11712963/reproducer.zip)

**Comment by shumonsharif** [comment]
Yes, you're right ... that section is for Maven and not Gradle specific.  However, the two key takeaways from that section that apply regardless of your build tool of choice:

- Use the `quarkus-cxf-bom`
- There are specific compatible versions of `quarkus-cxf-bom` based on which version of Quarkus you are on

I am a complete novice with Gradle, and I have no clue why the `jakarta.xml.ws:jakarta.xml.ws-api` transitive dependency is not getting pulled in, however, the configuration below should allow your reproducer to start up cleanly:

```gradle
    implementation(enforcedPlatform("io.quarkiverse.cxf:quarkus-cxf-bom:1.5.14"))
    implementation("io.quarkiverse.cxf:quarkus-cxf")
    implementation("jakarta.xml.ws:jakarta.xml.ws-api")
```

**Comment by DMaxter** [comment]
I got this error now on the auto generated classes: 
```
package javax.jws does not exist
```

I googled around but the package sugested `jaxws-rt` doesn't seem to exist

**Comment by ppalaga** [comment]
@DMaxter could you please share a mimal reproducer?

**Comment by DMaxter** [comment]
Here is the new reproducer: 
[reproducer.zip](https://github.com/quarkiverse/quarkus-cxf/files/11728590/reproducer.zip)
I just added [a calculator wsdl](https://ecs.syr.edu/faculty/fawcett/Handouts/cse775/code/calcWebService/Calc.asmx) to the previous example (with the previous fix)

**Comment by DMaxter** [comment]
Were you able to reproduce the error I obtained?

**Comment by shumonsharif** [comment]
Hi @DMaxter, it's not the ideal solution, but can you check if the below helps?

```java
    implementation(enforcedPlatform("io.quarkiverse.cxf:quarkus-cxf-bom:1.5.14"))
    implementation("io.quarkiverse.cxf:quarkus-cxf")
    implementation("jakarta.xml.ws:jakarta.xml.ws-api")
    implementation("jakarta.jws:jakarta.jws-api")
```

**Comment by shumonsharif** [comment]
@DMaxter Did the suggestion above work for you?

**Comment by DMaxter** [comment]
For my project that raised a cycle, I'm currently working on a reproducer

**Comment by shumonsharif** [comment]
I hope you're not getting the same cyclic issue here https://github.com/quarkiverse/quarkus-cxf/issues/860?  Reproducer would be great regardless.

**Comment by DMaxter** [comment]
Sorry for the late response.
I'm having a very similar issue to that one. There seems to be a conflict between the database package and cxf.
I'm using MySQL Reactive connector and got a similar error.
Here is the reproducer: 
[reproducer.zip](https://github.com/quarkiverse/quarkus-cxf/files/11926195/reproducer.zip)

**Comment by ppalaga** [comment]
#860 is fixed in 2.2.2 and 2.3.0. It would perhaps be good to port the fix also to 1.5

**Comment by ppalaga** [comment]
> https://github.com/quarkiverse/quarkus-cxf/issues/860 is fixed in 2.2.2 and 2.3.0. It would perhaps be good to port the fix also to 1.5

Done in https://github.com/quarkiverse/quarkus-cxf/pull/968

Giben that the original issue seems to have been resolved by @shumonsharif, I am closing this one. @DMaxter please feel free to re-open if you feel something is missing.
